src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+            return false;
+        }
+
+        // check if integer is within bounds
+        if (taskId >= taskList.size() || taskId < 0) {
+            System.out.println("☹ OOPS!!! That is an invalid task id.");
+            return false;
+        }
+
+        Task task = taskList.get(taskId);
+        task.markAsDone();
+
+        System.out.println("Nice! I've marked this task as done: \n"
+                + task.getStatusIcon() + " "
+                + task.getDescription()
+        );
+
+        return false;
+    }
+
+    /**
+     * Executes Todo Command
+     *
+     * @param argument  command arguments
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeTodoCommand(String argument) {
+        if (argument == null) {
+            System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+            return false;
+        }
+
+        Task task = new Todo(argument);
+        taskList.add(task);
+
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task.getTypeIcon() + task.getStatusIcon() + " "
+                + task.getDescription());
+        System.out.println("Now you have " + taskList.size() + " tasks in the list");
+        return false;
+    }
+
+    /**
+     * Executes Deadline Command
+     *
+     * @param argument  command arguments
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeDeadlineCommand(String argument) {
+        if (argument == null) {
+            System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+
+        String[] split = argument.split("/by", 2);
+        String description = split[0];
+        String by = split.length > 1 ? split[1].strip() : null;
+        Task task = new Deadlines(description, by);
+        taskList.add(task);
+
+        System.out.println("Got it. I have added this task:");
+        System.out.println("  " + task.getTypeIcon() + task.getStatusIcon() + " "
+                + task.getDescription());
+        System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        return false;
+    }
+
+    /**
+     * Executes Events Command
+     *
+     * @param argument  command arguments
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeEventCommand(String argument) {
+        if (argument == null) {
+            System.out.println("☹ OOPS!!! The description of an event cannot be empty.");
+            return false;
+        }
+
+        String[] split = argument.split("/at", 2);
+        String description = split[0];
+        String at = split.length > 1 ? split[1] : null;
+        Task task = new Events(description, at);
+        taskList.add(task);
+
+        System.out.println("Got it. I have added this task:");
```

Comment:
Perhaps these bunch of lines can be abstracted out too as well. They appear repeated in the other executeXXXXCommand methods as mentioned above.

Reply:
Thank you for the suggestion!